src/Client.php

```diff
@@ -1256,6 +1256,11 @@ protected function prepareHeaders($body, $uri)
                 $headers['Content-Length'] = strlen($body);
             }
         }
+        else {
+            if ($this->getMethod() == 'POST') {
+                $headers['Content-Length'] = 0;
```

Comment:
What about `PUT`, `PATCH`, etc.?

Also, this kind of change must be accompanied with a test.

Reply:
Hi,

I'm not a developer but I can try to improve this PR. Do you have a procedure to run phpunit test locally ?

Reply:
Hey,

You can do this locally, if you have `php` and `composer` installed:

```sh
git clone <this-repo>
composer install
./vendor/bin/phpunit
```

Reply:
The [contributing guidelines](https://github.com/laminas/.github/blob/e6a512745785a1f183c92d41cbb39bbaa3fda717/CONTRIBUTING.md) can also be helpful.

Reply:
Hello,

I had a test. 

For PUT, PATCH and others, I'm not sure what to do...

Reply:
I did some tests and I observe the same behavior for PUT so I did the same fix.